### ISSUE TITLE
Improve support for using svgcleaner as a Library.

### ISSUE THREAD

**Issue by leesdolphin** [RazrFalcon/svgcleaner#74]
I would like to use svgcleaner as part of a larger program, however attempting to interact with the library outside of running the command is very difficult.

It would be preferable for library users to call the individual cleaning functions; however if that is not desirable, then a simple change to `svgcleaner::cleaner::clean_doc` to allow it to take a `Vec<svgcleaner::cli::Key>` instead of a `clap::ArgMatches` object.

**Comment by RazrFalcon** [comment]
Well, I can do both, but it still will be a Rust library. No C-API are planned.

There are few problems with a direct access to cleaning options:
 - `clean_doc` contains so called [preclean checks](https://github.com/RazrFalcon/svgcleaner/blob/master/src/cleaner.rs#L52..L63) and [postclean checks](https://github.com/RazrFalcon/svgcleaner/blob/master/src/cleaner.rs#L195..L201)
 - order of cleaning functions is important

Basically, I can do both: `clean_doc` with an Options struct + low-level api for all cleaning methods.

**Comment by RazrFalcon** [comment]
Done. [Example](https://github.com/RazrFalcon/svgcleaner/blob/master/examples/use_as_lib/src/main.rs).

Remember that library is still under GPLv2.

**Comment by leesdolphin** [comment]
That looks really cool, thanks for doing that. I will have to have a try tomorrow.
